src/hypothesis/extra/django/models.py

```diff
@@ -128,6 +130,17 @@ def _get_strategy_for_field(f):
             min_size=(None if f.blank else 1),
             max_size=f.max_length,
         )
+        # We can infer a vastly more precise strategy by considering the
+        # validators as well as the field type.  This is a minimal proof of
+        # concept, but we intend to leverage the idea much more heavily soon.
+        # See https://github.com/HypothesisWorks/hypothesis-python/issues/1116
+        for v in f.validators:
+            if isinstance(v, validators.RegexValidator) and \
+                    not v.inverse_match:
+                if isinstance(v.regex, str):
```

Comment:
Currently if there are multiple `RegexValidator`s this picks the last one in the list. I don't have a problem with that, but it would be nice to add a comment indicating that this is an edge case that may not work that well.

If you'd rather deal with it than comment, one option in this case would be to use `one_of(from_regex(a), from_regex(b), ...)` on those. It'll still be an exact match because we filter by the validators later, but will deal with the case where the results of one regex are much more likely to match than the results of the other. Up to you if you care enough to want to deal with that.

Reply:
I'll add a comment.  For future work, the `one_of` tactic sounds good to me.

Alternatively, we could interpret the regexes as DFAs and compute their intersection - if anyone wants to write this (based on eg. https://qntm.org/greenery) they'd be welcome, but I'm not doing it.  The slog to support a subset of regex woudl be utterly *epic*...

Reply:
I've actually got a largish amount of prototype-quality code that does just this sort of thing (e.g https://travis-ci.org/DRMacIver/FALBS).

I would say that I am like 80% confident that with judicious use of laziness and regular expression derivatives I could make an automaton based solution that incorporates both `from_regex` and intersection given, say, two solid weeks of work.

There are number of amusing obstacles to overcome (like the fact that Python regex are actually context-free languages, so the resulting decision problem of deciding whether the intersection of two "regex" has a match is undecidable. But that's OK, we can work around that pretty easily because we only have to be better than generate + filter we don't have to be perfect), but I've got a pretty good idea of how they can all be overcome.

The biggest obstacle however is that I don't actually care enough to want to spend two solid weeks of work on this. :-p It might be worth revisiting if and when I've added a bunch more internal machinery for this though, which I'd say currently has some chance of happening given my current research direction.